### PR TITLE
Francesco better validation

### PR DESCRIPTION
**Pull request by Gamaranto**
Adds Better Validation to the Proposals Forms

### CODE REVIEW

File: packages/joy-proposals/src/validationSchema.ts [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+      .max(COUNCIL_SIZE_MAX, errorMessage("The council size", COUNCIL_SIZE_MIN, COUNCIL_SIZE_MAX))
+  },
+  Spending: {
+    tokens: Yup.number()
+      .positive("The token amount should be positive.")
+      .max(TOKENS_MAX, errorMessage("The amount of tokens", TOKENS_MIN, TOKENS_MAX))
+      .required("You need to specify an amount of tokens."),
+    destinationAccount: Yup.string()
+      .required("Select a destination account!")
+      .test("address-test", "${account} is not a valid address.", account => !!checkAddress(account, 5))
```
Comment (Lezek123) [inline packages/joy-proposals/src/validationSchema.ts:225]: Not sure if it's possible to provide invalid address in `InputAddress` field

File: packages/joy-proposals/src/validationSchema.ts [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+        MAX_VALIDATOR_COUNT_MAX,
+        errorMessage("The max validator count", MAX_VALIDATOR_COUNT_MIN, MAX_VALIDATOR_COUNT_MAX)
+      )
+  },
+  SetStorageRoleParameters: {
+    min_stake: Yup.number()
+      .required("All parameters are required")
+      .positive("The minimum stake should be positive.")
+      .max(MIN_STAKE_MAX, errorMessage("Minimum stake", MIN_STAKE_MIN, MIN_STAKE_MAX, "JOY")),
+    min_actors: Yup.number()
```
Comment (Lezek123) [inline packages/joy-proposals/src/validationSchema.ts:259]: I think for most numbers (ie. this one) we also need to prevent decimals (like `0.001`)

Comment (Gamaranto) [reply packages/joy-proposals/src/validationSchema.ts:266]: Good catch. Fixed by adding  a `.integer()` where needed.

File: packages/joy-proposals/src/validationSchema.ts (hunk 1 above)
Comment (Gamaranto) [reply packages/joy-proposals/src/validationSchema.ts:231]: I think the Component we use takes care of it, but I'm not 100% sure so I've added the test anyway

File: packages/joy-proposals/src/validationSchema.ts [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+    startup_grace_period: Yup.NumberSchema<number>;
+    entry_request_fee: Yup.NumberSchema<number>;
+  };
+};
+
+const Validation: ValidationType = {
+  All: {
+    title: Yup.string()
+      .required("Title is required!")
+      .max(TITLE_MAX_LENGTH, "Title max length is 40 characters."),
```
[the remark this answers is not in the document]
Comment (Gamaranto) [reply packages/joy-proposals/src/validationSchema.ts]: Fixed it

**Pull request merged**